vividus-extension-selenium/src/main/java/org/vividus/bdd/monitor/AbstractScreenshotOnFailureMonitor.java

```diff
@@ -37,26 +32,30 @@
 import org.vividus.reporter.event.AttachmentPublishEvent;
 import org.vividus.reporter.model.Attachment;
 import org.vividus.selenium.IWebDriverProvider;
-import org.vividus.selenium.screenshot.IScreenshotTaker;
+import org.vividus.selenium.screenshot.Screenshot;
 import org.vividus.softassert.event.AssertionFailedEvent;
-import org.vividus.ui.context.IUiContext;
 
-public class ScreenshotOnFailureMonitor extends NullStepMonitor
+public abstract class AbstractScreenshotOnFailureMonitor extends NullStepMonitor
 {
     private static final String NO_SCREENSHOT_ON_FAILURE_META_NAME = "noScreenshotOnFailure";
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(ScreenshotOnFailureMonitor.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractScreenshotOnFailureMonitor.class);
 
     private final ThreadLocal<Boolean> takeScreenshotOnFailureEnabled = InheritableThreadLocal.withInitial(
         () -> Boolean.FALSE);
 
-    @Inject private EventBus eventBus;
-
-    @Inject private IBddRunContext bddRunContext;
+    private final EventBus eventBus;
+    private final IBddRunContext bddRunContext;
+    private final IWebDriverProvider webDriverProvider;
 
-    @Inject private IWebDriverProvider webDriverProvider;
-    @Inject private IUiContext uiContext;
-    @Inject private IScreenshotTaker screenshotTaker;
+    public AbstractScreenshotOnFailureMonitor(EventBus eventBus, IBddRunContext bddRunContext,
+            IWebDriverProvider webDriverProvider)
+    {
+        this.eventBus = eventBus;
+        this.bddRunContext = bddRunContext;
+        this.webDriverProvider = webDriverProvider;
+        this.eventBus.register(this);
```

Comment:
why is it needed?

Reply:
SubscriberRegisteringBeanPostProcessor doesn't support finding of subscribers in parent classes, this is should be separate commit

Reply:
let's fix this issue first?

Reply:
in progress

Reply:
Will rebase after https://github.com/vividus-framework/vividus/pull/961